packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3014,6 +3029,34 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     ) {
         this.verifyNotClosed();
         assert(this.connected, 0x259 /* "Container disconnected when trying to submit system message" */);
+
+        if (new TextEncoder().encode(JSON.stringify(contents)).length > 2000) {
```

Comment:
Is there a better way to check the size of the payload here?

Reply:
Before the calls to submitRuntimeMessage we have teh serializedContent.length . Could we add it as an argument and avoid the length here ?